libs/shared/contentful/src/lib/factories.ts

```diff
@@ -23,6 +29,79 @@ import {
 } from './queries/services-with-capabilities';
 import { ContentfulErrorResponse } from './types';
 
+export const CapabilityServiceByPlanIdsQueryFactory = (
+  override?: Partial<CapabilityServiceByPlanIdsQuery>
+): CapabilityServiceByPlanIdsQuery => {
+  return {
+    purchaseCollection: {
+      items: [
+        {
+          stripePlanChoices: [faker.string.sample()],
+          offering: {
+            capabilitiesCollection: {
+              items: [
+                {
+                  slug: faker.string.sample(),
+                  servicesCollection: {
+                    items: [
+                      {
+                        oauthClientId: faker.string.sample(),
+                      },
+                    ],
+                  },
+                },
+              ],
+            },
+          },
+        },
+      ],
+    },
+    ...override,
+  };
+};
+
+export const CapabilityOfferingResultFactory = (
+  override?: Partial<CapabilityOfferingResult>,
+  capabilitiesExtension?: CapabilityCapabilitiesResult[],
+  servicesExtension?: CapabilityServicesResult[]
```

Comment:
Only one override parameter should be provided. To facilitate this behavior, this factory should be broken up into multiple factories for each of the subtypes, and all consume eachother.